### ISSUE TITLE
[Bug]: Sourcmap is wrong when using pluginReact

### ISSUE THREAD

**Issue by Skizy** [web-infra-dev/rsbuild#3467]
### Version

```sh
System:
    OS: Linux 6.10 EndeavourOS
    CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor
    Memory: 5.07 GB / 15.57 GB
    Container: Yes
    Shell: 5.9 - /bin/zsh
  npmPackages:
    @rsbuild/core: ^1.0.1 => 1.0.2 
    @rsbuild/plugin-react: ^1.0.1 => 1.0.1
```

### Details

This is the default sourcemap after creating a project with `bun create rsbuild@latest` and choosing React and TypeScript
![rsbuild_sourcemap](https://github.com/user-attachments/assets/ae6d4632-36ba-4a8b-8e60-ac2f866058d4)

It looks like the sourcemap is generated after one of the build steps
Doesn't happen with Solid, Svelte or Vue (haven't checked the rest)

### Reproduce link

default starter project

### Reproduce Steps

`bun create rsbuild@latest` or `npm create rsbuild@latest`
Choose React
Choose TypeScript

**Comment by Skizy** [comment]
Tested with JavaScript instead of TypeScript, the issue remains

**Comment by chenjiahan** [comment]
The same as https://github.com/web-infra-dev/rspack/issues/7870

We will release Rspack v1.0.5 today to fix this.

**Comment by Skizy** [comment]
Fixed in 1.0.3 (plugin-react 1.0.2)
